project/Deps.sc

```diff
@@ -531,7 +533,7 @@ object Izumi {
     artifacts = Seq(
       Artifact(
         name = Projects.docs.microsite,
-        libs = (cats_all ++ zio_all ++ http4s_all ++ doobie).map(_ in Scope.Compile.all),
+        libs = (cats_all ++ zio_all ++ http4s_all ++ doobie ++ List(monix_bio)).map(_ in Scope.Compile.all),
```

Comment:
@VladPodilnyk Hmm, why does microsite require monix-bio now?

Reply:
@VladPodilnyk But that means that optional dependency is not actually optional

Reply:
@VladPodilnyk Could you add the method that produces the error to OptionalDependencyTest and check what happens there?

Reply:
@neko-kai, this test passes, but I expect to catch `NullPointerException` here (like with `BIOConvert3to2` described below):
```scala
 "test " in {
    def adder[F[+_, +_]: BIOMonad: BIOPrimitives](i: Int) = BIOPrimitives[F].mkRef(0).flatMap(ref => ref.update(_ + i) *> ref.get)

    implicit val bioMonad: BIOMonad[Identity2] = null
    implicit val primitives: BIOPrimitives[Identity2] = null
    assert(adder[Identity2](0) == 0)
  }
```

BIOConver3to2 test:
```scala
def threeTo2[FR[-_, +_, +_]](implicit FR: BIO3[FR]): FR[Any, Nothing, Unit] = {
      val F: BIO[FR[Any, +?, +?]] = implicitly // must use `BIOConvert3To2` instance to convert FR -> F
      F.unit
    }

    type EitherFn[-R, +E, +A] = R => Either[E, A]
    implicit val bioEitherFn: BIO3[EitherFn] = null
    try threeTo2[EitherFn]
    catch { case _: NullPointerException => }
```

Reply:
also, such test  passes
```scala
assertCompiles(
      """import izumi.functional.bio.{F, BIOMonad, BIOMonadAsk, BIOPrimitives, BIORef3}
        |
        |def adder[F[+_, +_]: BIOMonad: BIOPrimitives](i: Int): F[Nothing, Int] =
        |  F.mkRef(0)
        |   .flatMap(ref => ref.update(_ + i) *> ref.get)
        |""".stripMargin)
```

Reply:
this assert passes
```scala
assertCompiles("""import izumi.functional.bio.{F, BIOMonad, BIOMonadAsk, BIOPrimitives, BIORef3}
                     |
                     |def adder[F[+_, +_]: BIOMonad: BIOPrimitives](i: Int): F[Nothing, Int] =
                     |  F.mkRef(0)
                     |   .flatMap(ref => ref.update(_ + i) *> ref.get)
                     |""".stripMargin)
```